examples/text/amazonreviewpolarity.py

```diff
@@ -37,25 +32,29 @@ def AmazonReviewPolarity(root, split):
     """Demonstrating caching, extraction and sanity check pipelines."""
 
     url_dp = IterableWrapper([URL])
-    # cache data on-disk with sanity check
-    cache_dp = url_dp.on_disk_cache(
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-
-    cache_dp = FileOpener(cache_dp, mode="b")
-
-    # stack TAR extractor on top of loader DP
-    extracted_files = cache_dp.read_from_tar()
-
-    # filter files as necessary
-    filter_extracted_files = extracted_files.filter(lambda x: split in x[0])
-
-    # stack sanity checker on top of extracted files
-    check_filter_extracted_files = filter_extracted_files.check_hash(
-        {os.path.normpath(os.path.join(root, _EXTRACTED_FILES[split])): _EXTRACTED_FILES_MD5[split]},
-        "md5",
-    )
-
-    # stack CSV reader and do some mapping
-    return check_filter_extracted_files.parse_csv().map(fn=lambda t: (int(t[0]), t[1]))
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    def extracted_filepath_fn(x):
+        file_path = os.path.join(root, _EXTRACTED_FILES[split])
+        dir_path = os.path.dirname(file_path)
```

Comment:
@ejguan Another thing I realized while working is end_caching won't be able to save file to disk if the corresponding parent directory doesn't exist. I have to explicitly create it. Do you think it would make sense to upstream directory creation in torchdata. For instance `tar` do the same and create appropriate directories when extracting files, I guess it would be non-intuitive for users to explicitly create directories for end_caching?

Reply:
As you can see from the code https://github.com/pytorch/data/blob/c06066ae360fc6054fb826ae041b1cb0c09b2f3b/torchdata/datapipes/iter/util/saver.py#L39-L40, `Saver` would create corresponding parent directory.

Reply:
got it, I might be something wrong. Just checked-in the new code, and yes it does not require to explicitly create the directory.